source/PlayServicesResolver/src/PlayServicesResolver.cs

```diff
@@ -97,6 +97,13 @@ internal class DependencyState {
             public void WriteToFile() {
                 try {
                     Directory.CreateDirectory(Path.GetDirectoryName(DEPENDENCY_STATE_FILE));
+                    if (!FileUtils.CheckoutFile(DEPENDENCY_STATE_FILE))
+                    {
+                        logger.Log(
+                            String.Format("Unable to checkout '{0}'. Dependency state has not been saved!",
+                                DEPENDENCY_STATE_FILE), LogLevel.Error);
```

Comment:
The problem here is that if this fails to save and auto-resolution is enabled, the resolver will get stuck in a loop attempting to resolve since the set of dependencies in memory won't match those in the state file.  I'll merge and clean fix this up.

Reply:
I see, I’ve actually seen that before and is one of the reasons we don’t enable auto resolution.

I think that if this file doesn’t save then it should probably fail resolution. I was unsure how to achieve this as I couldn’t return a failure from inside that function.